### ISSUE TITLE
azure resource set does not allow for PATCH operations

### ISSUE THREAD

**Issue by bmoore-msft** [Azure/azure-xplat-cli#2571]
setting properties on a resource can be done with PUT or PATCH, resource types that require PATCH cannot be done with the CLI (or at least I couldn't figure out how) because it seems to always do a PUT.  PowerShell has a switch to allow for PATCH, CLI should probably have one as well.

for example:

`azure resource set vaults myvault "microsoft.keyvault/vaults" -p $properties 2015-06-01 
`

fails because KeyVault expects a PATCH.

**Comment by amarzavery** [comment]
@ravbhatnagar, @vivsriaus - This is more of a service API design question. Can you please respond to this?

**Comment by vivsriaus** [comment]
Indeed, the current ARM resource command doesn't support PATCH operations. We'll track this as a work item on our end. @ravbhatnagar please add this to trello as well.

**Comment by amarzavery** [comment]
@ravbhatnagar @vivsriaus - Any update or timeline to support this in ResourceManager?

**Comment by ravbhatnagar** [comment]
Unfortunately, no timeline on this one too as yet.
